### ISSUE TITLE
Inconsistency between the paper and source code

### ISSUE THREAD

**Issue by qiuzhen8484** [thebestannie/Detection-free-MICCAI2023#1]
Hello, that's a solid work. After I went through the source code, I found that clustering and aggregate steps are absent in the model forward function. What is the reason for it?

**Comment by thebestannie** [comment]
Thank you for your attention, we adopted two stages in our previous work (under about 5000 subject samples) , and during our rebuttal submission time, we continued our new experiments under about 10000 subject samples, and we find the gap between coarse stage and fine stage is very small. Due to the longer inference time of two stages, we adopted just coarse stage in our real product, and we are also refining our pipeline with subsequence experiments. Actually, the checkpoint in this repo is training after 10000 subject samples. If you would like to adopt two stages method as in the paper, we can submit to this repo after reorganize the repo. By the way, we have conducted a new experiment under 50000 subject samples, and will release in https://github.com/bittersweet1999/cervical-cancer. Welcome if you have more question.

**Comment by qiuzhen8484** [comment]
> Thank you for your attention, we adopted two stages in our previous work (under about 5000 subject samples) , and during our rebuttal submission time, we continued our new experiments under about 10000 subject samples, and we find the gap between coarse stage and fine stage is very small. Due to the longer inference time of two stages, we adopted just coarse stage in our real product, and we are also refining our pipeline with subsequence experiments. Actually, the checkpoint in this repo is training after 10000 subject samples. If you would like to adopt two stages method as in the paper, we can submit to this repo after reorganize the repo. By the way, we have conducted a new experiment under 50000 subject samples, and will release in https://github.com/bittersweet1999/cervical-cancer. Welcome if you have more question.

Thanks for your reply. Fewer inference time is fascinating. In this regard, is it because the network is randomly initialized that more training samples are needed? In addition, the input for the network is 224 resized from 1024. Would it result a too small cell nucleus for diagnosis?

**Comment by thebestannie** [comment]
Thanks for your question, actually it is not because the network randomly initialize (we fine-tuned our model based on previous models in our subsequence experiments), it is because we do have large amount of training samples, they are collect from related company, we adopted our pipeline on our product with doctor to do final check, in this way we can collected our data. In our previous experiments, we use small amount of data with cell-level labels to train our models, but doctors don't want to do dirty works such as cell-level labeling, so we just have the sample-level pos/neg label, thus comes the pipeline. And for your second question, we also considered this problem before our experiments, but after we tried, the model works, especially for the high-level positive samples such as HISL, ASC-H, resize won't hurt this.

**Comment by qiuzhen8484** [comment]
Have you tried the resolution 768 or larger?

**Comment by thebestannie** [comment]
You may omit a problem. if you adopted larger resolution, of course the results will be better, but due to the limit of  GPU memory, you can't load too many images of one subject, and it is fatal. For example, if you adopt 768 resolution, you may just load 20 images for one subject, but a WSI contains 300-500 images, the 20 images may not have pos cells.

**Comment by qiuzhen8484** [comment]
How do you select this 20 images for a WSI?

**Comment by thebestannie** [comment]
In our implement of 224 resolution, we select the images based on the image size (we can assume if an image have more cells, the blank white area is less, and the size of this images is larger). So we can sort images, and select top k of them. And also, in our previous experiments, we use a detection model to select these top k images.

**Comment by thebestannie** [comment]
Feel free to contack us if you have any other problems, and welcome to follow the new repo https://github.com/bittersweet1999/cervical-cancer, we will update our newset pipeline and model ckpts on it.
